src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -101,36 +126,242 @@ CodeAnalysis.Syntax.InternalSyntax.SyntaxList<InterpolatedStringContentSyntax> g
+                // definitely at the start of a newline (or at the end).  Regardless, we want to consume each successive
+                // line, making sure it's indentation is correct.
+
+                SyntaxDiagnosticInfo error = null;
+                while (currentIndex < text.Length)
+                {
+                    currentLineWhitespace.Clear();
+                    var lineStartPosition = currentIndex;
+                    while (currentIndex < text.Length && SyntaxFacts.IsWhitespace(text[currentIndex]))
+                    {
+                        currentLineWhitespace.Append(text[currentIndex]);
+                        currentIndex++;
+                    }
+
+                    // Only bother reporting a single error on a text chunk.
+                    if (error == null)
+                    {
+                        var isAtEndOfLastLine = last && currentIndex == text.Length;
+                        var isAtNewLine = currentIndex < text.Length && SyntaxFacts.IsNewLine(text[currentIndex]);
+                        if (isAtEndOfLastLine || isAtNewLine)
+                        {
+                            // a whitespace-only content line.  The indentation whitespace must be a prefix of the current line whitespace,
+                            // or vice versa.  It is an error otherwise.
+                            if (!Lexer.StartsWith(indentationWhitespace, currentLineWhitespace) &&
+                                !Lexer.StartsWith(currentLineWhitespace, indentationWhitespace))
+                            {
+                                error = MakeError(
+                                    lineStartPosition,
+                                    width: currentIndex - lineStartPosition,
+                                    ErrorCode.ERR_LineDoesNotStartWithSameWhitespace);
+                            }
+                        }
+                        else
+                        {
+                            // a content line with non-whitespace.  The indentation whitespace must be a prefix of the current line
+                            // whitespace.  It is an error otherwise.
+                            if (!Lexer.StartsWith(currentLineWhitespace, indentationWhitespace))
+                            {
+                                error ??= MakeError(
+                                    lineStartPosition,
+                                    width: currentIndex - lineStartPosition,
+                                    ErrorCode.ERR_LineDoesNotStartWithSameWhitespace);
+                            }
+                        }
+                    }
+
+                    // Skip the leading whitespace that matches the terminator line and add any whitespace past that to the
+                    // string value.
+                    for (var i = indentationWhitespace.Length; i < currentLineWhitespace.Length; i++)
+                        content.Append(currentLineWhitespace[i]);
```

Comment:
This can be done as a single `Append` as well, and maybe `indentationWhitespace` can be created as a `string`/`char[]`/`ReadOnlySpan<char>` without a StringBuilder.

Reply:
Then `Lexer.StartsWith` might stay private if `ReadOnlySpan<char>.StartsWith` or `String.StartsWith` ends up being what is used.

Reply:
yup.  much nicer!